make_env.py

```diff
@@ -89,7 +91,7 @@ def _parse_profile(profile_str: str, opt_deps: dict, project_name: str) -> List[
     else:
         conda_deps.append(dep)
 if args.py_version is not None:
-    conda_deps.append(f"python~={args.py_version}")
+    conda_deps.append(f"python~={args.py_version}.0")
```

Comment:
From SO I understand that e.g. `~=3.9.0` means `>=3.9.0, ==3.9.*` And before it went wrong because without the final `.0` it means `>=3.9, ==3.*`?

It seems less confusing to me to simply use `==3.9` which also gives you the latest 3.9 release right? 

Either way, as long as it works, I'm fine with it.

Reply:
I think @dalmijn  is also using the make_env.py in the hydromt_wflow repos and it is also in the cookie cutter. It would be good to update it in all these places to avoid future bugs.

Reply:
mostly correct I think. I remember looking into this before, and the reason we didn't use `==3.9` is because then you get `3.9.0` which does not include security updates. That's at least how I understand it, but honestly I find the whole thing very confusing. So I suggest we keep it like this for now as I've verified that this gives you the latest 3.9, but I'll look a bit more into it to see if I can understand it better